arch/arm/core/aarch32/fatal.c

```diff
@@ -17,6 +17,9 @@
 #include <logging/log.h>
 LOG_MODULE_DECLARE(os, CONFIG_KERNEL_LOG_LEVEL);
 
+/* global callee_saved regs, preserved before the exception system call */
+_callee_saved_t z_callee_saved;
```

Comment:
I notice that 'z_callee_saved' hasn't been used when CONFIG_EXTRA_EXCEPTION_INFO=n. It's better to define 'z_callee_saved' only when CONFIG_EXTRA_EXCEPTION_INFO=y.